main.go

```diff
@@ -0,0 +1,45 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+// Package main is the entrypoint for the sqlcmd CLI application.
+package main
+
+import (
+	"github.com/microsoft/go-sqlcmd/cmd"
+	legacyCmd "github.com/microsoft/go-sqlcmd/cmd/sqlcmd"
+	"os"
+)
+
+// main is the entrypoint function for sqlcmd.
+//
+// TEMPORARY: While we have both the new cobra and old kong CLI
+// implementations, main decides which CLI framework to use
```

Comment:
i suspect some version of this code is permanent. We aren't likely to reproduce all the legacy switches in the new CLI are we? Perhaps we should use an environment variable to only allow modern mode where it uses `query` as the default command but only with whatever switches that command supports.

Reply:
I think we can solve the problem of the legacy switches by writing a small "canonicalizer" called at the start of main, which scans through the supplied args, and modifies them to be parsable by Cobra, then call cobra.command.SetArgs to override what's passed in from env.

I'll look at this staight after this PR is in.  I've entered an issue and assigned it to myself.

Reply:
Added to:

[https://github.com/microsoft/go-sqlcmd/issues/154](https://github.com/microsoft/go-sqlcmd/issues/154)